src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/MS/Internal/MarkupCompiler/MarkupCompiler.cs

```diff
@@ -2596,7 +2596,20 @@ private void GenerateInitializeComponent(bool isApp)
 
             string uriPart = string.Empty;
 
-            string version = String.IsNullOrEmpty(AssemblyVersion) ? String.Empty : COMPONENT_DELIMITER + VER + AssemblyVersion;
+            // If we have an actual version string, try to detect wildcards.  Note that this is required to be at least major.minor.
+            // Prior to markup compilation, this will have been validated by the C# compiler.
+            bool wildcardUsed = false;
+
+            if (!string.IsNullOrEmpty(AssemblyVersion))
+            {
+                var splitVersion = AssemblyVersion.Split('.');
+                wildcardUsed = splitVersion[splitVersion.Length - 1] == "*";
```

Comment:
You need index checks here. IIRC you mentioned that csc would do additional checks for us, but we can not rely on that for all languages. This needs to be robust for VB, as well as any future language support we might add. We should not rely on C# compilers behavior here - we should do our own independent validation that the version is well-formed

Reply:
VB is a good point.  I can do the full version check like Roslyn does, but we'll have to add errors and localization.  Let me look into adding all of that.

Reply:
Side note, the [VB compiler](https://github.com/dotnet/roslyn/tree/master/src/Compilers/VisualBasic) is under Roslyn as well.  It uses the same version checks.

Reply:
You might want to be robust against IL postprocessing tools editing the assembly version as well, depending on how they are embedded in the msbuild chain they may be running before you are looking at the attributes.

Reply:
That's why the note about robustness against future language support 😛. C++/CLI, F# etc. support are both plausible in near future; F# esp. seems to be growing in popularity and we've heard some requests. 

In this case, the validation isn't very hard to implement locally either, so IMHO we should just add it here. We could even borrow it from Roslyn.

Reply:
I already borrowed the Roslyn code and modified it for our purposes.

Resolving this against that.